index.js

```diff
@@ -6,58 +6,30 @@ export default function pLimit(concurrency) {
 	const queue = new Queue();
 	let activeCount = 0;
 
-	const resumeNext = () => {
+	const next = () => {
 		if (activeCount < concurrency && queue.size > 0) {
-			queue.dequeue()();
-			// Since `pendingCount` has been decreased by one, increase `activeCount` by one.
 			activeCount++;
+			queue.dequeue()();
 		}
 	};
 
-	const next = () => {
-		activeCount--;
+	const generator = async (function_, ...arguments_) => {
+		const dequeuePromise = new Promise(resolve => {
```

Comment:
Is there a reason for creating a Promise first?

For example, due to the queueMicrotask specification, I need to create the Promise first for it to work properly、.etc.

As far as I know, there is no problem with creating the Promise after the queueMicrotask, etc.

If you have a reason for this, please leave it in the comments.